plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -1152,6 +1152,8 @@ private static String escapeNamePattern(String name, String escape)
         name = name.replace(escape, escape + escape);
         name = name.replace("_", escape + "_");
         name = name.replace("%", escape + "%");
+        name = name.replace("[", escape + "[");
+        name = name.replace("]", escape + "]");
```

Comment:
I believe this is not needed for any SQL spec-conformant database.
It looks like something required for SQL Server to work correctly.
Am i correct?

Reply:
https://github.com/trinodb/trino/pull/14286#discussion_r979761331